internal/controller/pdbwatcher_controller.go

```diff
@@ -102,14 +122,39 @@ func (r *PDBWatcherReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 	deployment := &appsv1.Deployment{}
 	err = r.Get(ctx, types.NamespacedName{Name: deploymentName, Namespace: pdbWatcher.Namespace}, deployment)
 	if err != nil {
-		return ctrl.Result{}, err
+		return ctrl.Result{}, err // Error fetching Deployment
 	}
 
+	// Track initial state to detect conflicts
+	initialResourceVersion := deployment.ResourceVersion
+	initialReplicas := *deployment.Spec.Replicas
+
 	// Update status dynamically
-	pdbWatcher.Status.CurrentReplicas = *deployment.Spec.Replicas
-	pdbWatcher.Status.MinReplicas = *deployment.Spec.Replicas
-	pdbWatcher.Status.MaxReplicas = *deployment.Spec.Replicas + deployment.Spec.Strategy.RollingUpdate.MaxSurge.IntVal
-	pdbWatcher.Status.ScaleFactor = 1 // Dynamically Calculated
+	if pdbWatcher.Status.CurrentReplicas == 0 {
+		pdbWatcher.Status.CurrentReplicas = initialReplicas
+	}
+	pdbWatcher.Status.MinReplicas = initialReplicas
```

Comment:
Setting this here before checking resource version loses the deployents actual initial sate. Only want to reset this if someone changed this besides the controller (resource version is different than what is saved)

Reply:
You don't want to do this as you're going to lose information if you surged and get called again. THis is not the initial replicas its the current replicas from the deployment.